pkg/controller/servicebindingrequest/planner.go

```diff
@@ -44,15 +45,37 @@ func (p *Planner) searchCR(kind string) (*unstructured.Unstructured, error) {
 	logger.Info("Searching for CR instance...")
 
 	cr, err := p.client.Resource(gvr).Namespace(p.sbr.GetNamespace()).Get(bss.ResourceRef, opts)
+
 	if err != nil {
-		logger.Error(err, "during reading CR")
+		logger.Info("during reading CR")
 		return nil, err
 	}
 
 	logger.WithValues("CR.Name", cr.GetName()).Info("Found target CR!")
 	return cr, nil
 }
 
+// searchCRD based on a CustomResourceDefinitionDescription and name, search for the object.
+func (p *Planner) searchCRD() (*unstructured.Unstructured, error) {
+	bss := p.sbr.Spec.BackingServiceSelector
+	gvk := schema.GroupVersionKind{Group: "apiextensions.k8s.io", Version: "v1beta1", Kind: "CustomResourceDefinition"}
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	opts := metav1.GetOptions{}
+
+	logger := p.logger.WithValues("CR.GVK", gvk.String(), "CR.GVR", gvr.String())
+	logger.Info("Searching for CRD instance...")
+
+	crd, err := p.client.Resource(gvr).Namespace(p.sbr.GetNamespace()).Get(bss.Kind, opts)
```

Comment:
There is a difference in namespace. You created your mock in "planner" namespace -> https://github.com/redhat-developer/service-binding-operator/pull/183/files#diff-8187ea18445343ad6726c5be04e8b8d5R57

But `p.sbr.GetNamespace()` here gives me a different namespace ("testing").

```
crd, err := p.client.Resource(gvr).Namespace(p.sbr.GetNamespace()).Get(bss.Kind, opts)
```

Reply:
There is another test which pass the value as "testing". Can you please check that?
`pkg/controller/servicebindingrequest/retriever_test.go` (Line 260)